nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py

```diff
@@ -356,6 +356,78 @@ def batch_decode(self, ids):
     return tokenizer
 
 
+def load_nemo_config(nemo_ckpt: Union[str, Path]) -> Dict[Any, Any]:
+    """
+    Load the model configuration from a NeMo checkpoint.
+
+    This function handles both NeMo 1.0 and NeMo 2.0 checkpoint structures.
+    For NeMo 2.0, it reads the configuration from the 'context/model.yaml' file.
+    For NeMo 1.0, it uses the UnpackedNemoCheckpointDir to load the model configuration.
+
+    Args:
+        nemo_ckpt (Union[str, Path]): Path to the NeMo checkpoint file or directory.
+    Returns:
+        Dict[Any, Any]: The configuration dictionary.
+    """
+    if Path(nemo_ckpt).is_dir():
+        nemo_ckpt = Path(nemo_ckpt)
+    else:
+        nemo_ckpt = TarPath(nemo_ckpt)
+
+    if (nemo_ckpt / "weights").exists() and (nemo_ckpt / "context").exists():  # Stucture of NeMo 2.0 checkpoints
+        with (nemo_ckpt / "context" / "model.yaml").open("r") as stream:
+            config = yaml.safe_load(stream)
+    else:  # Assume NeMo 1.0 case
+        unpacked_checkpoint_dir = UnpackedNemoCheckpointDir(nemo_ckpt, load_checkpoints_to_cpu=True)
+        config = unpacked_checkpoint_dir.model_config
+
+    return config
+
+
+def get_model_type(nemo_ckpt: Union[str, Path]) -> Optional[str]:
+    """
+    Determine the model type from a NeMo checkpoint for TensorRT-LLM engine build.
+
+    Args:
+        nemo_ckpt (str): Path to the NeMo checkpoint file.
+    Returns:
+        Optional[str]: The model type if it can be determined, otherwise None.
+    """
+    model_config = load_nemo_config(nemo_ckpt)
+    model_type = None
+
+    if model_class := model_config.get("_target_"):
+        # NeMo 2.0 case
+        NEMO2_TO_MODEL_TYPE = {
+            "nemo.collections.llm.gpt.model.base.GPTModel": "gpt",
+            "nemo.collections.llm.gpt.model.llama.LlamaModel": "llama",
+            "nemo.collections.llm.gpt.model.mistral.MistralModel": "llama",
+            "nemo.collections.llm.gpt.model.mixtral.MixtralModel": "llama",
+            "nemo.collections.llm.gpt.model.starcoder.StarcoderModel": "gpt",
+            "nemo.collections.llm.gpt.model.starcoder2.Starcoder2Model": "gpt",
+            "nemo.collections.llm.gpt.model.nemotron.NemotronModel": "gpt",
```

Comment:
For NeMo 1.0 I use `gptnext`. Is it correct? Is `gptnext` used in NeMo 2.0?

Reply:
They can be used interchangeably, see for example, [here](https://github.com/NVIDIA/NeMo/blob/69322161339b9b348af65763669f629e2d6b68e4/nemo/export/tensorrt_llm.py#L433-L434).

But I suggest using "gpt" as this conforms also with recent ModelOpt update [0.19.0](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/release/0.19.0/llm_ptq/example_utils.py#L30-L57) -> [0.21.0](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/release/0.21.0/llm_ptq/example_utils.py#L30-L60)